File: src/main/java/com/ada/shared/infrastructure/DataSourceConfiguration.java
package com.ada.shared.infrastructure;

import java.nio.file.*;
import javax.sql.DataSource;
import org.springframework.context.annotation.*;
import org.sqlite.SQLiteDataSource;

@Configuration(proxyBeanMethods = false)
public class DataSourceConfiguration {
  @Bean
  public DataSource dataSource(AdaProperties p) throws Exception {
    Path d = p.getNormalizedDataDirectory().resolve("db");
    Files.createDirectories(d);
    var ds = new SQLiteDataSource();
    ds.setUrl("jdbc:sqlite:" + d.resolve("ada.sqlite"));
    return ds;
  }
}

File: src/main/kotlin/com/ada/shared/infrastructure/DataSourceConfiguration.kt
package com.ada.shared.infrastructure

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.sqlite.SQLiteDataSource
import javax.sql.DataSource

@Configuration(proxyBeanMethods = false)
class DataSourceConfiguration {
    @Bean
    fun dataSource(properties: AdaProperties): DataSource = SQLiteDataSource().apply {
        url = "jdbc:sqlite:${properties.normalizedDataDirectory.resolve("db/ada.sqlite")}"
    }
}
